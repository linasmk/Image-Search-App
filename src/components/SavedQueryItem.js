/* ========= App Dependencies ============= */
import React from "react";
import PropTypes from "prop-types";

/* ========= Code ============= */
export const SavedQueryItem = ({ removeSavedQuery, onClick, name, id }) => {
  /* ========== Functions ========== */
  const removeQuery = () => removeSavedQuery({ id });

  const handleChildQuery = (e) => {
    onClick(e.target.textContent);
  };

  /* ========== Content ========== */
  return (
    <div className="saved-queries__item">
      <p
        className="saved-queries__txt"
        role="presentation"
        onClick={handleChildQuery}>
        {name}
      </p>
      <button
        type="button"
        className="saved-queries__btn--remove"
        onClick={removeQuery}>
        &#88;
      </button>
    </div>
  );
};
SavedQueryItem.defaultProps = {
  onClick: undefined,
  removeSavedQuery: undefined
};
SavedQueryItem.propTypes = {
  removeSavedQuery: PropTypes.func,
  onClick: PropTypes.func,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};
export default SavedQueryItem;
