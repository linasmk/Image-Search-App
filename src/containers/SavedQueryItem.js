/* ===== Redux ===== */
import { connect } from "react-redux";
import { removeSavedQuery } from "@actions/action-creators";
/* ====== Components ===== */
import SavedQueryItemComponent from "@comp/SavedQueryItem";
/* ========= Code ========= */

export const mapDispatchToProps = (dispatch) => ({
  removeSavedQuery: (id) => dispatch(removeSavedQuery(id))
});

export const SavedQueryItem = connect(
  null,
  mapDispatchToProps
)(SavedQueryItemComponent);
export default SavedQueryItem;
