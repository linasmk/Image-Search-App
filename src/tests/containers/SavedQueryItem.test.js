import React from "react";
import { mount } from "enzyme";
import Root from "@tests/Root";
import SavedQueryItem, { mapDispatchToProps } from "@cont/SavedQueryItem";
import queries from "@tests/fixtures/queries";

describe("SavedQueryItem container", () => {
  it("renders SavedQueryItem container correctly", () => {
    const wrapper = mount(
      <Root>
        <SavedQueryItem
          removeSavedQuery={jest.fn()}
          onClick={jest.fn()}
          name="dog"
          id="1"
        />
      </Root>
    );
    console.log(wrapper);
    expect(wrapper).toMatchSnapshot();
  });
  describe("mapDispatchToProps", () => {
    it("should map dispatch to props correctly", () => {
      const query = queries[0].id;
      const queryId = "1";
      const removeSavedQuery = jest.fn();
      mapDispatchToProps(removeSavedQuery(query));

      expect(removeSavedQuery).toHaveBeenLastCalledWith(queryId);
      expect(removeSavedQuery).toHaveBeenCalledTimes(1);
    });
  });
});
