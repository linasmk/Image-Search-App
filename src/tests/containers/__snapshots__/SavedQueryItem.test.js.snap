// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedQueryItem container renders SavedQueryItem container correctly 1`] = `
<Root>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(SavedQueryItem)
      id="1"
      name="dog"
      onClick={[MockFunction]}
      removeSavedQuery={[MockFunction]}
    >
      <SavedQueryItem
        id="1"
        name="dog"
        onClick={[MockFunction]}
        removeSavedQuery={[Function]}
      >
        <div
          className="saved-queries__item"
        >
          <p
            className="saved-queries__txt"
            onClick={[Function]}
            role="presentation"
          >
            dog
          </p>
          <button
            className="saved-queries__btn--remove"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
      </SavedQueryItem>
    </Connect(SavedQueryItem)>
  </Provider>
</Root>
`;
